app/controllers/api/subcollections/tags.rb

```diff
@@ -84,11 +93,8 @@ def parse_tag(data)
       end
 
       def parse_tag_from_href(data)
-        href = data["href"]
-        tag  = if href && href.match(%r{^.*/tags/#{CID_OR_ID_MATCHER}$})
-                 klass = collection_class(:tags)
-                 klass.find(from_cid(href.split('/').last))
-               end
+        href = parse_href(data['href'])
```

Comment:
still some work left to do on this PR but I saw some room for refactoring / reuse which would negate the need for #13597. Thoughts on this approach @abellotti @imtayadeway?